marklogic-data-hub-central/ui/src/components/entities/mapping/create-edit-mapping-dialog/create-edit-mapping-dialog.tsx

```diff
@@ -136,6 +140,27 @@ const CreateEditMappingDialog = (props) => {
     </div>
   </Modal>;
 
+  const onSave = async (event) => {
```

Comment:
Looks like we can update the existing onSubmit function only with an if condition like given below and get rid of this onSave function. 
if(mapName && (collections || srcQuery)) {
 // Current logic within onSubmit
} else {
//Just set the corresponding '~touched' flags for the properties for example isMapNameTouched as true.
}

Reply:
I added all the changes you suggested.  Thanks for the feedback!